### DISCUSSION TITLE
October 2024 Survey Results - Question 9

### DISCUSSION THREAD

**Discussion by boc-the-git** [mealie-recipes/mealie#4647] [General]
Placeholder discussion for people to share their thoughts and insights about the survey results.

**Comment by matthewkdies** [comment]
First off, thank you all for everything you all have done -- Mealie is without a doubt the best app of many that I self host, and has elevated cooking from a casual interest to a passion to me! I’m endlessly grateful to all contributors and the open-source community, and look forward to contributing when the inspiration strikes.

---

I’m not sure if this is a suggestion or a request from other users for an improved workflow. Our household has shopping lists split by "meal type"; for example, we have a dinner list, lunch list, etc. This makes it easiest to add items into lists from a meal plan, so a dinner meal will go right onto the dinner list without having to enter it into multiple lists.

When shopping, this can be semi-frustrating. If there are fruit items on both the lunch and breakfast lists, we have to switch between both within the store repeatedly to check. We do tag every food item on the list and can sort within lists by tag to ease this, but it is still counterintuitive to switch between the lists.

My suggestion during the survey was the ability to create a "parent list" that could contain all items from children lists. It might be possible to create multiple, which could practically work as lists per store. For example, a Kroger list, a Meijer list, a Trader Joe's list, a Walmart list, etc. But even a singular parent list that could contain all items and be sortable by tags would be incredible!

Like I said, if there's a workflow I’m missing, I’m definitely open to it. But I think one or more parent or "store" lists would be incredible!

**Reply by boc-the-git** [reply]
@matthewkdies what is your use case for having a "lunch list" vs a "breakfast list"? The meal plan would be the source of truth for "when am I going to eat the food", the shopping list is just the "buy X and Y".
Wondering if there's a good reason I'm not thinking of.

**Reply by matthewkdies** [reply]
Our initial use case was to group items by like items before we knew about labels and parsing ingredients into foods. I suppose now we could pretty easily convert our lists into a list per store, and then use the label grouping and organization while shopping.

**Comment by matthewkdies** [comment]
Here are some other unrelated things that would really make the shopping experience excellent in my opinion:
1. ~~**Ability to order tags:** I would love the ability to change where tags are within a list. For example, the deli in my grocery store comes prior to the bakery. Ordering tags into the order in which I shop would be powerful!~~
2. **Item ordering "memory" or pre-creation:** I order items on my list within tags in order of the route I take through the store. If I could create a file that orders items (or do it through the GUI!) this would enable automatic sorting of items as they enter the list.
3. **Improved sorting usability:** Clicking and dragging items on lists to order them feels "unresponsive". I often have to click multiple times to grab the item to move it, and going beyond the size of the screen can cause scrolling that is hard to control.

**Comment by t-persson** [comment]
> Here are some other unrelated things that would really make the shopping experience excellent in my opinion:
> 1. **Ability to order tags:** I would love the ability to change where tags are within a list. For example, the deli in my grocery store comes prior to the bakery. Ordering tags into the order in which I shop would be powerful!

When you say "tags" do you mean "labels"?
I am running an older version of mealie, but in my shopping list i can reorder labels.

In order to show that button I click the "toggle label sort" button
![Screenshot_20241214_214001_Brave.jpg](https://github.com/user-attachments/assets/1cebe580-f1ed-4f5b-9c9a-1e1f6b5fdf25)

**Reply by matthewkdies** [reply]
You're absolutely right, I totally missed the big blue "Reorder Labels" button! Haha. Will edit the comment, thank you!

**Reply by ChipSkylark37** [reply]
Just recently started using Labels under Shopping List - Definitely helps when picking up things like 'Alcohol' or navigating different sections of the store.

**Comment by t-persson** [comment]
Running version v1.10.2 so if this feature already exists in newer versions, please ignore me.

A feature I would like for the shopping list is a "keep screen awake" toggle, similar to the one on the recipes. My phone often locks when running around the shop and its mildly annoying :)

**Reply by Kuchenpirat** [reply]
Shopping list wake lock was added in #4236

**Reply by t-persson** [reply]
> Shopping list wake lock was added in #4236 

Guess it's about time I upgrade then. Thanks!

**Reply by ChipSkylark37** [reply]
This doesn't work within the PWA for me, running iOS 18.2. I believe this is a known bug that Apple hasn't implemented into mobile safari. 

I now color code two Mealie Icons - One for PWA and one without so depending on what I need to do, I'll use one or another. Not ideal but it works.

**Comment by ChipSkylark37** [comment]
I wasn't able to fill out the survey but some that came to mind:

1. Integration into Apple Reminders.

Currently I have a lot of 'Location Based' reminders so that when I arrive Home, at a Store, etc, it reminds me to pick something up. This would be great to then get alerted if there's something on my Mealie list that they may have at that store. I find often that I may add an ingredient to pick up _the next time im at the store_ and then forget to actually check in Mealie. So, I may then add it to a location based reminder in Apple Reminders instead. A little cumbersome and I hate to have multiple lists to then check off, but it works. 

2. Easily Print / Print to PDF. 

For when I may want a physical copy to bring to the store. Some of the stores in my area have notoriously bad cell service / wifi coverage. In this instance, having a physical printed copy, or perhaps an **Offline Mode** would be beneficial. 

3. Prune Shopping Lists

I keep a single Shopping List Currently. Partially because it's what works, but more because I haven't played around enough to determine why I would need multiple. I thought of creating one per store or product type, but that seemed like too much to maintain. How I run it right now, it would be nice to then have an 'auto-prune' checked off items from the list, by some selectable date. Maybe after 1 week, 2 weeks, etc.

I know there is the option to 'delete checked items' but I would prefer to leave a few days worth, just in case I think "wait, what did I get recently?"

4. Undo Button "toast" notification.

I sometimes accidentally check off the wrong item. If there was a possibility to have a notification dialog popup to Undo that disappears after a few seconds, that would save me the clicks of expanding the Checked Items and undoing it. It could display for some user select value (1-3 seconds?) and then disappear. This would be better than an 'Are you sure' button so no further interaction is required if it was fully intended. 

5. iOS Shortcut / Siri Interaction to add ingredient to list.

Maybe this is already possible? I read about the feature of iOS shortcuts and admittedly need to read more about it. If there was a way to say "Hey Siri, add Eggs to my Mealie Shopping List" or "Run Shopping List Shortcut" which then has a sub-action to ask "What Ingredient do you want to add" and then waits for a response, and adds it to the Shopping List in Mealie. As of now, I'm mostly using Location Based Reminders for Apple Reminders (See item 1) but since recipes are in Mealie, it makes the most sense to retain the Shopping List in there.

EDIT:
So far I've not gotten the iOS shortcut to work. I believe it's due to that I have a reverse proxy configured with cloudflare. I'm unsure if that works with how the shortcut works and don't want to punch a hole into my instance for the API port. I'll keep reviewing to see if I can get this functional, but from the documentation it appears to be for importing just a new recipe, not for adding any ingredients to the shopping list. 

=====================

I am sure there are more, but these are the ones I think about when using the App and the small 'irks' or 'wow, it'd be cool if...' moments. 

I will read up more on other requests or comment if I have more ideas.

**Reply by michael-genson** [reply]
> 'auto-prune' checked off items from the list, by some selectable date. Maybe after 1 week, 2 weeks, etc.

It's kind of a hidden feature (it's not documented anywhere) but checked-off shopping list items are deleted daily, past the first 100 (with the least-recently-checked items being deleted first)

**Reply by matthewkdies** [reply]
> checked-off shopping list items are deleted daily, past the first 100 (with the least-recently-checked items being deleted first)

Is there a way to disable this? I use a list that has many items we buy repeatedly, and do not want to re-add them every time.

**Reply by michael-genson** [reply]
Currently there is no way to disable this

**Reply by ChipSkylark37** [reply]
Interesting use case: 

For commonly purchased ingredients, have a list that you keep to check / uncheck, that could also "uncheck" themselves after a given timeframe if not done manually. 

If I check off "apples" on 12/26, then a date to "uncheck" can be set, reminding me to pick them up again in a week/two/whatever, automatically.

**Reply by matthewkdies** [reply]
While this would work, it would also require creating a separate list for each store that we shop at. This would lead to flipping between lists when shopping, which isn't the best UX IMO.

Another option is to do the described experience for all of the checked items on the list, but if there are existent previously checked items, these could get lost when re-checking the items.

IMO, the best option would be to allow the user to disable the auto-deletion per list. I've never contributed to Mealie, but I might take a crack at this if it seems doable. I’m pretty lacking in relevant web-dev experience though :)
